packages/components/src/components/hds/table/index.ts

```diff
@@ -4,29 +4,110 @@
+import type { HdsTableThSortArgs } from './th-sort.ts';
+import type { HdsTableTrArgs } from './tr.ts';
+import {
+  HdsTableDensityValues,
+  HdsTableThSortOrderValues,
+  HdsTableVerticalAlignmentValues,
+} from './types.ts';
+import type {
+  HdsTableColumn,
+  HdsTableDensities,
+  HdsTableHorizontalAlignment,
+  HdsTableSelectableRow,
+  HdsTableSortingFunction,
+  HdsTableThSortOrder,
+  HdsTableVerticalAlignment,
+} from './types';
+
+export const DENSITIES: string[] = Object.values(HdsTableDensityValues);
+export const DEFAULT_DENSITY = HdsTableDensityValues.Medium;
+
+export const VALIGNMENTS: string[] = Object.values(
+  HdsTableVerticalAlignmentValues
+);
+export const DEFAULT_VALIGN = HdsTableVerticalAlignmentValues.Top;
+
+export interface HdsTableArgs {
+  Args: {
+    align?: HdsTableHorizontalAlignment;
+    caption?: string;
+    columns?: HdsTableColumn[];
+    density?: HdsTableDensities;
+    identityKey?: string;
+    isFixedLayout?: boolean;
+    isSelectable?: boolean;
+    isStriped?: boolean;
+    model: Record<string, unknown>;
+    onSelectionChange?: (selection: {
+      selectionKey: string;
+      selectionCheckboxElement:
+        | HdsFormCheckboxBaseSignature['Element']
+        | undefined;
+      selectedRowsKeys: string[];
+      selectableRowsStates: {
+        selectionKey: string;
+        isSelected: boolean | undefined;
+      }[];
+    }) => void;
+    onSort?: (sortBy: string, sortOrder: HdsTableThSortOrder) => void;
+    selectionAriaLabelSuffix?: string;
+    sortBy?: string;
+    sortedMessageText?: string;
+    sortOrder?: HdsTableThSortOrder;
+    valign?: HdsTableVerticalAlignment;
+  };
+  Blocks: {
+    head?: [
+      {
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        ThSort: ComponentLike<HdsTableThSortArgs>;
+        sortBy: string | undefined;
+        sortOrder: HdsTableThSortOrder;
+        setSortBy: (column: string) => void;
+      },
+    ];
+    body?: [
+      {
+        Td: ComponentLike<HdsTableTdArgs>;
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        ThSort?: ComponentLike<HdsTableThSortArgs>;
+        data?: string;
+        sortBy?: string | undefined;
+        sortOrder?: HdsTableThSortOrder;
+      },
+    ];
+  };
+  Element: HTMLTableElement;
+}
 
-export const DENSITIES = ['short', 'medium', 'tall'];
-const DEFAULT_DENSITY = 'medium';
-const VALIGNMENTS = ['top', 'middle', 'baseline'];
-const DEFAULT_VALIGN = 'top';
-
-export default class HdsTableIndexComponent extends Component {
+export default class HdsTableIndexComponent extends Component<HdsTableArgs> {
   @tracked sortBy = this.args.sortBy;
-  @tracked sortOrder = this.args.sortOrder || 'asc';
-  @tracked selectAllCheckbox = undefined;
-  selectableRows = [];
-  @tracked isSelectAllCheckboxSelected = undefined;
+  @tracked sortOrder = this.args.sortOrder || HdsTableThSortOrderValues.Asc;
+  @tracked selectAllCheckbox:
```

Comment:
I think this could be

`@tracked selectAllCheckbox?: HdsFormCheckboxBaseSignature['Element'] = undefined`

Reply:
It could be, but I like using a Union Type with undefined because it explicitly states the possible types, making it clear that undefined is a valid value. But might be a good topic for engineering sync, if we want to define best practices around what we use shorthand for and what we do not.